nixos/modules/services/web-servers/nginx/default.nix

```diff
@@ -1224,8 +1246,8 @@ in
         # New file permissions
         UMask = "0027"; # 0640 / 0750
         # Capabilities
-        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ];
-        CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ];
+        AmbientCapabilities = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ] ++ optionals cfg.enableQuicBPF [ "CAP_SYS_ADMIN" "CAP_NET_ADMIN" ];
+        CapabilityBoundingSet = [ "CAP_NET_BIND_SERVICE" "CAP_SYS_RESOURCE" ] ++ optionals cfg.enableQuicBPF [ "CAP_SYS_ADMIN" "CAP_NET_ADMIN" ];
```

Comment:
Those are quite powerful capabilities. Does nginx drop them in its webserver?

Reply:
Without these features, eBPF-enabled nginx will not work :( . Nginx does not use CAP_BPF.

Reply:
@arut is it possible to implement use of eBPF only with Capability "CAP_BPF"?

Reply:
> Does nginx drop them in its webserver?

I've run getcaps on a worker process and it does not.

```
 ➜ getpcaps 11070
11070: cap_net_bind_service,cap_net_admin,cap_sys_admin,cap_sys_resource,cap_bpf=eip
```

Reply:
If this is how people like to run their webserver these days, so it be. Let's at least put a warning in.... next up we are going to modprobe a nginx.ko into the kernel for more performance.